src/components/home/balance-card.tsx

```diff
@@ -33,27 +33,43 @@ const BalanceCard = () => {
    * Fetch user's flextime data when it is undefined.
    */
   useEffect(() => {
-    if (!usersFlextime) {
+    // This card shows admin links if adminMode is true, no personal flextime for the admin themselves.
+    // So, only fetch if not adminMode and flextime data is not yet available.
+    if (!adminMode && !usersFlextime) {
       getUsersFlextimes();
+    } else if (adminMode) {
+      // If in admin mode, we are not displaying personal flextime, so ensure loading is false.
+      setLoading(false);
     }
-  }, [users]);
+    // Adding usersFlextime to dependencies to prevent re-fetch if data is already loaded.
+    // Adding adminMode to control fetching based on user role.
+    // loggedInUser and severaUserId are derived from users and userProfile, which are in the original deps or implied.
```

Comment:
Lets remove these comments.
Do we really need users and userProfile in the deps list? Does it still work as intended if they are removed?